rust/zypp-agama/src/lib.rs

```diff
@@ -413,6 +497,15 @@ mod tests {
         Ok(())
     }
 
+    /*
+    #[test]
+    fn init_target_deadlock() -> Result<(), Box<dyn Error>> {
```

Comment:
This test would simply deadlock. I wonder if it's worth it to use a library with mutexes that can time out?

Reply:
well, I would say it will be better if 
1. we use try_lock instead of lock and check for WoudlBlock errror - https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.try_lock
2. init_target return error if there is already initialized target.
3. add also close_target as we would need it when switching from out root to "/mnt"

Reply:
1. OK, `try_lock` works, thanks for the tip
2. for now I delay and `panic!`, or do you think Agama would benefit from a graceful error handler?
3. `close_target` is still TODO

Reply:
regarding 2. I think if we use the approach @imobachgs using we do not benefit much as we keep it in single thread and changing of target is really single step which we need when switch to /mnt